src/discolight/augmentations/randomshear.py

```diff
@@ -0,0 +1,39 @@
+"""An augmentation to randomly rotate an image."""
+import random
+import math
+from discolight.params.params import Params
+from .augmentation.types import Augmentation, NumericalRange
+from .shear import Shear
+from .decorators.accepts_probs import accepts_probs
+
+
+@accepts_probs
+class RandomShear(Augmentation):
+
+    """Randomly shear the given image."""
+
+    def __init__(self, shear_range):
+        """Construct a RandomShear augmentation.
+
+        You should probably use the augmentation factory or Discolight
+        library interface to construct augmentations. Only invoke
+        this constructor directly if you know what you are doing.
+        """
+        super().__init__()
+
+        self.shear_range = shear_range
+
+    @staticmethod
+    def params():
+        """Return a Params object describing constructor parameters."""
+        return Params().add(
+            "shear_range",
+            "The shear range has no bounds",
+            NumericalRange(-math.inf, math.inf), (0.2, 0.2))

```

Comment:
```suggestion
            NumericalRange(), (0.2, 0.2))

```

In general, you should use `None` instead of `±math.inf`. Note that if there is no minimum or maximum you can simply call the constructor with no arguments, like I've done here in the suggestion.